---
title: "Landfill methanogenic taxa metaanalyses"
output: html_notebook
editor_options: 
  chunk_output_type: inline
---
# Workflow
1. Load in the packages you'll need to conduct the data analyses
2. Update the taxonomy compiled from the literature to reflect the GTDB r89 database used in the manuscript
3. Create the presence absence figure from the compiled methanogic taxa and validate use of trimmed input data to reproduce figure

# Software and package versions
- Ran the sessionInfo() command on 2022-12-06 following Rstudio and R package updates

R version 4.1.2 (2021-11-01)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Monterey 12.3.1

Matrix products: default
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_CA.UTF-8/en_CA.UTF-8/en_CA.UTF-8/C/en_CA.UTF-8/en_CA.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] lubridate_1.9.0   timechange_0.1.1  readxl_1.4.1      glue_1.6.2        patchwork_1.1.2   ggtext_0.1.2      data.table_1.14.6 forcats_0.5.2    
 [9] stringr_1.5.0     dplyr_1.0.10      purrr_0.3.5       readr_2.1.3       tidyr_1.2.1       tibble_3.1.8      ggplot2_3.4.0     tidyverse_1.3.2  

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.9          assertthat_0.2.1    digest_0.6.30       utf8_1.2.2          R6_2.5.1            cellranger_1.1.0    backports_1.4.1    
 [8] reprex_2.0.2        evaluate_0.18       httr_1.4.4          pillar_1.8.1        rlang_1.0.6         googlesheets4_1.0.1 rstudioapi_0.14    
[15] jquerylib_0.1.4     rmarkdown_2.18      textshaping_0.3.6   labeling_0.4.2      googledrive_2.0.0   gridtext_0.1.5      munsell_0.5.0      
[22] broom_1.0.1         compiler_4.1.2      modelr_0.1.10       xfun_0.35           systemfonts_1.0.4   pkgconfig_2.0.3     htmltools_0.5.3    
[29] tidyselect_1.2.0    fansi_1.0.3         crayon_1.5.2        tzdb_0.3.0          dbplyr_2.2.1        withr_2.5.0         commonmark_1.8.1   
[36] grid_4.1.2          jsonlite_1.8.4      gtable_0.3.1        lifecycle_1.0.3     DBI_1.1.3           magrittr_2.0.3      scales_1.2.1       
[43] cli_3.4.1           stringi_1.7.8       cachem_1.0.6        farver_2.1.1        fs_1.5.2            xml2_1.3.3          bslib_0.4.1        
[50] ellipsis_0.3.2      ragg_1.2.4          generics_0.1.3      vctrs_0.5.1         RColorBrewer_1.1-3  tools_4.1.2         markdown_1.4       
[57] hms_1.1.2           fastmap_1.1.0       yaml_2.3.6          colorspace_2.0-3    gargle_1.2.1        rvest_1.0.3         knitr_1.41         
[64] haven_2.5.1         sass_0.4.4  

#1. Load R packages required for data visualization
```{r}
library("readxl")
library("tidyverse")
library("lubridate")
```

#2. Update the taxonomy from the literature to reflect GTDB r89
You need to normalize everything from the literature to GTDB taxonomical hierarchy. First, you need to convert the deepest taxonomic classification to its current name in GTDB since previous studies use older databases and NCBI more often than not. You only need to detect the name once in the input data to convert it.

*GTDB Version used in this study:* R89, The earliest version code that has consistent naming at the newly encoded GTDB taxonomy variable is indicated after the #.

**Note:** I worked by searching for a representative genome with genus name then manually going through GTDB's hierarchal diagrams to r86.2 of GTDB. From there, I filtered down to family rank to see if it changed in subsequent releases such as r89 used in this study.

1. Read the excel file in containing the raw compiled data.
```{r}
Methanogen_taxa<-read_excel("Methanogen_taxa_compilation.xlsx", sheet = "Methanogens")
```

2. Create a data frame containing distinct taxa names at all levels to work from, omitting the current study because the family variable is from GTDB. 
```{r}
Methanogen_genus_distinct<- Methanogen_taxa %>% filter(!Reference == "This_study") %>% distinct(Genus_reported)
Potential_non_methanogens_genus<- Methanogen_genus_distinct %>% filter(!grepl("Methano", Genus_reported))

Methanogen_family_distinct<- Methanogen_taxa %>% filter(!Reference == "This_study") %>% distinct(Family_reported)
Potential_non_methanogens_family<- Methanogen_family_distinct %>% filter(!grepl("Methano", Family_reported))
```

Typos to correct:
- Methanosarci**n**ca --> Methanosarcina

Genera to follow up for omission after searching for the prefix "Methano" in compiled data:
- Thermodiscus--> Genus in Desulfurococcaceae, not methanogenic to the best of my knowledge
- Pyrobaculum
- Halorhabdus
- Methanimicrococcus --> Methanosarcinaceae, according to NCBI, genus doesn't show up until R95 for GTDB in Methanosarcinaceae family
- Thermocladium
- Candidatus_Parvarchaeum
- Halostagnicola
- Natrinema
- Candidatus_Nitrosopumilus
- Ferroplasma
- Thermoplasma

3. Working from the genus level, update the families reported following GTDB hierarchy. Follow a similar workflow to convert family names to GTDB r89 and update families reported based on species name as well. 
```{r}
Methanogen_taxa_GTDB <- Methanogen_taxa %>%
                        mutate(Family_GTDB = case_when(
                        
                        #genus to family conversion  
                        Genus_reported == "Methanimicrococcus" ~ "Methanosarcinaceae", #only appears in R95
                        Genus_reported == "Methanobacterium" ~ "Methanobacteriaceae", #back to R86.2
                        Genus_reported == "Methanobrevibacter" ~ "Methanobacteriaceae", #back to R86.2
                        Genus_reported == "Methanocalculus" ~ "Methanocorpusculaceae", #back to R86.2
                        Genus_reported == "Methanocaldococcus" ~  "Methanocaldococcaceae", #back to R86.2
                        Genus_reported == "Methanocella" ~ "Methanocellaceae", #back to R86.2
                        Genus_reported == "Methanocorpusculum" ~ "Methanocorpusculaceae", #back to R86.2
                        Genus_reported == "Methanoculleus" ~ "Methanocullaceae", #back to R86.2
                        Genus_reported == "Methanogenium" ~ "Methanomicrobiaceae", #back to R86.2
                        Genus_reported == "Methanofollis" ~ "Methanofollaceae", #back to R86.2
                        Genus_reported == "Methanolinea" ~ "Methanoregulaceae", #back to R86.2
                        Genus_reported == "Methanomassiliicoccus" ~ "Methanomassiliicoccaceae", #back to R86.2
                        Genus_reported == "Methanoplanus" ~ "Methanomicrobiaceae", #back to R86.2, lone genus
                        Genus_reported == "Methanoregula" ~ "Methanoregulaceae", #back to R86.2
                        Genus_reported == "Methanosarcina" ~ "Methanosarcinaceae", #back to R86.2
                        Genus_reported == "Methanosarcinca" ~ "Methanosarcinaceae", #back to R86.2
                        Genus_reported == "Methanosaeta" ~ "Methanotrichaceae", #back to R86.2
                        Genus_reported == "Methanosphaerula" ~ "Methanosphaerulaceae", #back to R86.2, lone genus
                        Genus_reported == "Methanospirillum" ~ "Methanospirillaceae", #back R86.2
                        Genus_reported == "Methanothermobacter" ~ "Methanothermobacteriaceae", #back to R86.2
                        Genus_reported == "Methanomethylophilus" ~ "Methanomethylophilaceae", #back to 86.2
                        Genus_reported == "Candidatus Methanomethylophilus" ~ "Methanomethylophilaceae", #back to 86.2
                        Genus_reported == "Methanomicrobium" ~ "Methanomicrobiaceae", #sole genus, back to 86.2
                        Genus_reported == "Methanolobus" ~ "Methanosarcinaceae", #sole genus, back to 89
                        Genus_reported == "Methanofastidiosum" ~ "Methanofastidiosaceae", #back to 86.2
                        Genus_reported == "Methanothrix" ~ "Methanotrichaceae", #back to 86.2
                        
                        #family conversions
                        Family_reported == "Methanobacteriaceae" ~ "Methanobacteriaceae",
                        Family_reported == "Methanosarcinaceae" ~ "Methanosarcinaceae",
                        Family_reported == "Methanomicrobiaceae" ~ "Methanomicrobiaceae",
                        Family_reported == "Methanosaetaceae" ~ "Methanotrichaceae", #recoded based on manual GTDB search for Methanoseta genus
                        Family_reported == "Methanosarcinacea" ~ "Methanosarcinaceae", #corrected for typo
                        Family_reported == "Methanosarcinaceae" ~ "Methanosarcinaceae",
                        Family_reported == "Methanomethylophilaceae" ~"Methanomethylophilaceae",
                        
                        #species to family conversions
                        grepl("Methanosaeta", Species_reported) ~ "Methanotrichaceae", #following same rules for genus as above
                        grepl("Methanosarcina", Species_reported) ~"Methanosarcinaceae", #following same rules for genus as above
                        grepl("Methanomethylovorans", Species_reported) ~ "Methanosarcinaceae", #back to R86.2
                        Reference == "This_study" ~ Family_reported)) %>%
                        mutate(Family_GTDB = na_if(Family_GTDB, "NA"))
```

# 3. Create the presence absence figure 
1. Create a theme for the figure here.
```{r}
heatmap_theme<- theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
               panel.background = element_blank(), axis.line = element_line(colour = "black"),
               axis.text.x = element_text(color = "black", size = 14),
               axis.text.y = element_text(color = "black", size = 14),
               axis.title.x = element_text(color = "black", size = 16, vjust = -3),
               axis.title.y = element_text(color = "black", size = 16, vjust = 6),
               legend.title=element_text(size=16), 
               legend.text=element_text(size=14),
               legend.key = element_rect(fill = "white", color = NA),
               plot.title = element_text(size = 18, face = "bold", hjust = 0.5),
               panel.border = element_rect(colour = "black", fill=NA, size=1),
               plot.margin = unit(c(0, 0, 0, 0), "cm"))
```

2. Generate a figure that includes reformatted references that were abbreviated manually 
```{r}
#Generate counts for how frequently the families were detected across the dataset
Reference_count<-Methanogen_taxa_GTDB %>% 
                 drop_na(Family_GTDB) %>% #this is where you filter out taxa if need be
                 filter(!Original_sample_ID == "CSWMC") %>%
                 group_by(Family_GTDB) %>%
                 summarise(count = n_distinct(Reference)) #this gets us the most frequently occurring families


Split_presence_absence_heatmap<- Methanogen_taxa_GTDB %>% 
                                 left_join(Reference_count) %>%
                                 drop_na(Family_GTDB) %>% #this is where you filter out taxa if need be
                                 filter(!Original_sample_ID == "CSWMC") %>%
                                 mutate(Data_source = case_when(
                                   Reference == "This_study" ~ "This_study",
                                   !Reference == "This_study" ~"Literature")) %>%
                                 mutate(Reference = case_when(
                                   Original_sample_ID == "A" ~ "A (39 years old)",
                                   Original_sample_ID == "B" ~ "B (37 years old)",
                                   Original_sample_ID == "C" ~ "C (31 years old)",
                                   Original_sample_ID == "D1" ~ "D1 (26 years old)",
                                   Original_sample_ID == "D2" ~ "D2 (24 years old)",
                                   Original_sample_ID == "E" ~ "E (20 years old)",
                                   Original_sample_ID == "F1" ~ "F1 (5 years old)",
                                   Original_sample_ID == "F2" ~ "F2 (3 years old)",
                                   !Data_source == "This_study" ~ Reference)) %>%
  
#reformat references
mutate(Reference = recode(Reference, 'Sauk_Hug_2022' = "A.H. Sauk & L.A. Hug, ISME Commun.  **2**, 58 (2022)")) %>%
mutate(Reference = recode(Reference, 'De_La_Cruz_et_al_2021' = "F. De la Cruz, *et al.* Waste Management **124**, 26-35 (2021)")) %>%
mutate(Reference = recode(Reference, 'Wan_et_al_2021' = "S. Wan, *et al.* Diversity-Basel **13**, 6 (2021)")) %>%
mutate(Reference = recode(Reference, 'Thakur_et_al_2020' = "K. Thakur, *et al.* World. J. Microbiol. Biotechnol. **36**, 3 (2020)")) %>%
mutate(Reference = recode(Reference, 'Sillas_Moreno_et_al_2019' = 
                                     "M. Sillas-Moreno, *et al.* J. Chem. Technol. Biotechnol. **94**, 4, 1123-1133 (2019)")) %>%
mutate(Reference = recode(Reference, 'Staley_et_al_2018' = "B. Staley, *et al.* Appl. Microbiol. Biotechnol. **102**, 13, 5731-5740 (2018)")) %>%
mutate(Reference = recode(Reference, 'Stamps_et_al_2016' = "B. Stamps, *et al.* Front. Microbiol. **7**, 534 (2016)")) %>%
mutate(Reference = recode(Reference, 'Tang_et_al_2016' = "W. Tang, *et al.* FEMS Microbiol. Lett. **363**, 9, 1-7 (2016)")) %>%
mutate(Reference = recode(Reference, 'Dong_et_al_2015' = "J. Dong, *et al.* Appl. Biochem. Biotechnol. **175**, 5, 2729-2740 (2015)")) %>%
mutate(Reference = recode(Reference, 'Fei_et_al_2015' = "X. Fei, *et al.* Waste Management **36**, 184-190 (2015)")) %>%
mutate(Reference = recode(Reference, 'Song_et_al_2015' = "L. Song, *et al.* Appl. Microbiol. Biotechnol. **99**, 14, 6125-6137 (2015)")) %>%
mutate(Reference = recode(Reference, 'Yadav_et_al_2015' = "S. Yadav, *et al.* Archaea **2015**, 1-9 (2015)")) %>%
mutate(Reference = recode(Reference, 'Bareither_et_al_2013' = "C. Bareither, *et al.* Waste Management **33**, 10, 1982-1992 (2013)")) %>%
mutate(Reference = recode(Reference, 'Qu_et_al_2009' = "X. Qu, *et al.* FEMS Microbiol. Ecol. **68**, 2, 236-245 (2009)")) %>%
mutate(Reference = recode(Reference, 'Laloui_carpentier_et_al_2006' = "W. Laloui-Carpentier *et al.* Antonie van Leeuwenhoek **89**, 3-4, 423-434 (2006)")) %>%
mutate(Reference = recode(Reference, 'Chen_et_al_2003a' = "A. Chen, *et al.* Biotechnol. Lett. **25**, 18, 1563-1569 (2003)")) %>%
mutate(Reference = recode(Reference, 'Huang_et_al_2003' = "L. Huang, *et al.* FEMS Microbiol. Ecol.  **46**, 2, 171-177 (2003)")) %>%
mutate(Reference = recode(Reference, 'Uz_et_al_2003' = "I. Uz, *et al.* Proc. R. Soc. London, Ser. B. **270**, Suppl. 2, S202-S205 (2003)")) %>%
mutate(Reference = recode(Reference, 'Luton_et_al_2002' = "P. Luton, *et al.* Microbiology **148**, 11, 3521-3530 (2002)"))

#split the heatmap into different facets to reflect this study vs data compiled from the literature
Split_presence_absence_heatmap$Data_source_ordered = factor(Split_presence_absence_heatmap$Data_source,
                                                            levels=c('This_study','Literature'))
#plot the figure
Split_presence_absence_heatmap<- Split_presence_absence_heatmap %>%
                                 mutate(Data_source_ordered = 
                                          recode(Data_source_ordered, 'This_study' = "This study")) %>%
                                 mutate(Family_GTDB = str_replace(string = Family_GTDB,
                                                pattern = "(.*)",
                                                replacement = "*\\1*")) %>%
                                 mutate(Sequencing_method = str_replace(string = Sequencing_method,
                                                pattern = "mcrA",
                                                replacement = "*mcrA*")) %>%
                                 ggplot(aes(x= reorder(Reference, -Year_published), y = reorder(Family_GTDB, count),
                                      fill = Sequencing_method)) +
                                 geom_tile(aes()) + #add the width argument here
                                 facet_grid(~Data_source_ordered, scales = "free") +
                                 theme(axis.text.x=element_text(angle=90, hjust=1)) +
                                 #scale_fill_manual(values = c("#009933"))+
                                 labs(title = "Methanogen families detected in landfill studies", 
                                      x = "Reference", 
                                      y = "Family", 
                                      fill = "Method of detection") + 
                                 heatmap_theme + 
                                 theme(plot.title = element_text(hjust = 0.5, vjust = -1),
                                       legend.position = "right",
                                       strip.text.x = element_text(size = 18),
                                       axis.text.y = element_markdown(size = 16, color = "black"),
                                       axis.text.x = element_markdown(size = 12, color = "black", angle = 45),
                                       legend.text = element_markdown(size = 16, color = "black"))

Split_presence_absence_heatmap
ggsave("Fig4_Methanogen_metaanalyses.pdf", height=12, width=18, device="pdf")
```
3. Output a trimmed down table as a .tsv for reproducibility.
```{r}
Methanogen_taxa_GTDB_trimmed <- Methanogen_taxa_GTDB %>%
                                select(-c("Domain_reported":"Order_reported")) %>%
                                select(-c("Comment", "Main_points_discussion")) %>%
                                left_join(Reference_count) %>%
                                 drop_na(Family_GTDB) %>% #this is where you filter out taxa if need be
                                 filter(!Original_sample_ID == "CSWMC") %>%
                                 mutate(Data_source = case_when(
                                   Reference == "This_study" ~ "This_study",
                                   !Reference == "This_study" ~"Literature")) %>%
                                 mutate(Reference = case_when(
                                   Original_sample_ID == "A" ~ "A (39 years old)",
                                   Original_sample_ID == "B" ~ "B (37 years old)",
                                   Original_sample_ID == "C" ~ "C (31 years old)",
                                   Original_sample_ID == "D1" ~ "D1 (26 years old)",
                                   Original_sample_ID == "D2" ~ "D2 (24 years old)",
                                   Original_sample_ID == "E" ~ "E (20 years old)",
                                   Original_sample_ID == "F1" ~ "F1 (5 years old)",
                                   Original_sample_ID == "F2" ~ "F2 (3 years old)",
                                   !Data_source == "This_study" ~ Reference)) %>%
mutate(Reference = recode(Reference, 'Sauk_Hug_2022' = "A.H. Sauk & L.A. Hug, ISME Commun.  **2**, 58 (2022)")) %>%
mutate(Reference = recode(Reference, 'De_La_Cruz_et_al_2021' = "F. De la Cruz, *et al.* Waste Management **124**, 26-35 (2021)")) %>%
mutate(Reference = recode(Reference, 'Wan_et_al_2021' = "S. Wan, *et al.* Diversity-Basel **13**, 6 (2021)")) %>%
mutate(Reference = recode(Reference, 'Thakur_et_al_2020' = "K. Thakur, *et al.* World. J. Microbiol. Biotechnol. **36**, 3 (2020)")) %>%
mutate(Reference = recode(Reference, 'Sillas_Moreno_et_al_2019' = 
                                     "M. Sillas-Moreno, *et al.* J. Chem. Technol. Biotechnol. **94**, 4, 1123-1133 (2019)")) %>%
mutate(Reference = recode(Reference, 'Staley_et_al_2018' = "B. Staley, *et al.* Appl. Microbiol. Biotechnol. **102**, 13, 5731-5740 (2018)")) %>%
mutate(Reference = recode(Reference, 'Stamps_et_al_2016' = "B. Stamps, *et al.* Front. Microbiol. **7**, 534 (2016)")) %>%
mutate(Reference = recode(Reference, 'Tang_et_al_2016' = "W. Tang, *et al.* FEMS Microbiol. Lett. **363**, 9, 1-7 (2016)")) %>%
mutate(Reference = recode(Reference, 'Dong_et_al_2015' = "J. Dong, *et al.* Appl. Biochem. Biotechnol. **175**, 5, 2729-2740 (2015)")) %>%
mutate(Reference = recode(Reference, 'Fei_et_al_2015' = "X. Fei, *et al.* Waste Management **36**, 184-190 (2015)")) %>%
mutate(Reference = recode(Reference, 'Song_et_al_2015' = "L. Song, *et al.* Appl. Microbiol. Biotechnol. **99**, 14, 6125-6137 (2015)")) %>%
mutate(Reference = recode(Reference, 'Yadav_et_al_2015' = "S. Yadav, *et al.* Archaea **2015**, 1-9 (2015)")) %>%
mutate(Reference = recode(Reference, 'Bareither_et_al_2013' = "C. Bareither, *et al.* Waste Management **33**, 10, 1982-1992 (2013)")) %>%
mutate(Reference = recode(Reference, 'Qu_et_al_2009' = "X. Qu, *et al.* FEMS Microbiol. Ecol. **68**, 2, 236-245 (2009)")) %>%
mutate(Reference = recode(Reference, 'Laloui_carpentier_et_al_2006' = "W. Laloui-Carpentier *et al.* Antonie van Leeuwenhoek **89**, 3-4, 423-434 (2006)")) %>%
mutate(Reference = recode(Reference, 'Chen_et_al_2003a' = "A. Chen, *et al.* Biotechnol. Lett. **25**, 18, 1563-1569 (2003)")) %>%
mutate(Reference = recode(Reference, 'Huang_et_al_2003' = "L. Huang, *et al.* FEMS Microbiol. Ecol.  **46**, 2, 171-177 (2003)")) %>%
mutate(Reference = recode(Reference, 'Uz_et_al_2003' = "I. Uz, *et al.* Proc. R. Soc. London, Ser. B. **270**, Suppl. 2, S202-S205 (2003)")) %>%
mutate(Reference = recode(Reference, 'Luton_et_al_2002' = "P. Luton, *et al.* Microbiology **148**, 11, 3521-3530 (2002)")) %>%
                                select(-c("count")) %>%drop_na(Family_GTDB) %>% relocate(Data_source)
  

write.table(Methanogen_taxa_GTDB_trimmed, file = "FileS4_input_data_methanogenic_taxa_compilation.tsv", sep ='\t', row.names = FALSE)
```

4. Confirm the visualization can be reproduced using the trimmed tab-delimited input file.
```{r}
Split_presence_absence_heatmap<-fread("FileS4_input_data_methanogenic_taxa_compilation.tsv", sep = "\t", header = "auto")

Reference_count<-Split_presence_absence_heatmap %>% 
                 drop_na(Family_GTDB) %>% #this is where you filter out taxa if need be
                 filter(!Original_sample_ID == "CSWMC") %>%
                 group_by(Family_GTDB) %>%
                 summarise(count = n_distinct(Reference)) 

Split_presence_absence_heatmap<- Split_presence_absence_heatmap %>% left_join(Reference_count)

#split the heatmap into different facets to reflect this study vs data compiled from the literature
Split_presence_absence_heatmap$Data_source_ordered = factor(Split_presence_absence_heatmap$Data_source,
                                                            levels=c('This_study','Literature'))
#plot the figure
Split_presence_absence_heatmap<- Split_presence_absence_heatmap %>%
                                 mutate(Data_source_ordered = 
                                          recode(Data_source_ordered, 'This_study' = "This study")) %>%
                                 mutate(Family_GTDB = str_replace(string = Family_GTDB,
                                                pattern = "(.*)",
                                                replacement = "*\\1*")) %>%
                                 mutate(Sequencing_method = str_replace(string = Sequencing_method,
                                                pattern = "mcrA",
                                                replacement = "*mcrA*")) %>%
                                 ggplot(aes(x= reorder(Reference, -Year_published), y = reorder(Family_GTDB, count),
                                      fill = Sequencing_method)) +
                                 geom_tile(aes()) + #add the width argument here
                                 facet_grid(~Data_source_ordered, scales = "free") +
                                 theme(axis.text.x=element_text(angle=90, hjust=1)) +
                                 labs(title = "Methanogen families detected in landfill studies", 
                                      x = "Reference", 
                                      y = "Family", 
                                      fill = "Method of detection") + 
                                 heatmap_theme + 
                                 theme(plot.title = element_text(hjust = 0.5, vjust = -1),
                                       legend.position = "right",
                                       strip.text.x = element_text(size = 18),
                                       axis.text.y = element_markdown(size = 16, color = "black"),
                                       axis.text.x = element_markdown(size = 12, color = "black", angle = 45),
                                       legend.text = element_markdown(size = 16, color = "black"))

Split_presence_absence_heatmap
ggsave("Fig4_Methanogen_metaanalyses_from_trimmed_input.pdf", height=12, width=18, device="pdf")
```

