---
title: "Nevskiaceae phylogenomic analysis to screen for methanotrophy potential"
Date: '2022-12-20'
output:
  html_notebook:
    theme: united
    toc: yes
  html_document:
    toc: yes
    df_print: paged
---


# Workflow
1. Use GToTree to create a rooted tree for Methanoperedenacaeae genomes with an outgroup
2. Visualize phylogenetic trees in R
3. Run the Methanoperedenaceae genomes available from GTDB through the DRAM workflow
4. Reproduce the heatmaps from DRAM in R

# Software and package versions
- Ran the sessionInfo() command on 2022-12-21 following Rstudio and R package updates

R version 4.1.2 (2021-11-01)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Monterey 12.3.1

Matrix products: default
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_CA.UTF-8/en_CA.UTF-8/en_CA.UTF-8/C/en_CA.UTF-8/en_CA.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] tidytree_0.4.1   aplot_0.1.9      ggnewscale_0.4.8 treeio_1.16.2    ggtree_3.0.4     glue_1.6.2       forcats_0.5.2    stringr_1.5.0    dplyr_1.0.10    
[10] purrr_0.3.5      readr_2.1.3      tidyr_1.2.1      tibble_3.1.8     ggplot2_3.4.0    tidyverse_1.3.2 

loaded via a namespace (and not attached):
 [1] httr_1.4.4          sass_0.4.4          jsonlite_1.8.4      modelr_0.1.10       bslib_0.4.1         assertthat_0.2.1    yulab.utils_0.0.5  
 [8] googlesheets4_1.0.1 cellranger_1.1.0    yaml_2.3.6          pillar_1.8.1        backports_1.4.1     lattice_0.20-45     digest_0.6.30      
[15] rvest_1.0.3         colorspace_2.0-3    ggfun_0.0.9         htmltools_0.5.4     pkgconfig_2.0.3     broom_1.0.1         haven_2.5.1        
[22] patchwork_1.1.2     scales_1.2.1        ggplotify_0.1.0     tzdb_0.3.0          timechange_0.1.1    googledrive_2.0.0   farver_2.1.1       
[29] generics_0.1.3      ellipsis_0.3.2      cachem_1.0.6        withr_2.5.0         lazyeval_0.2.2      cli_3.4.1           magrittr_2.0.3     
[36] crayon_1.5.2        readxl_1.4.1        evaluate_0.18       fs_1.5.2            fansi_1.0.3         nlme_3.1-160        xml2_1.3.3         
[43] textshaping_0.3.6   tools_4.1.2         hms_1.1.2           gargle_1.2.1        lifecycle_1.0.3     munsell_0.5.0       reprex_2.0.2       
[50] compiler_4.1.2      jquerylib_0.1.4     systemfonts_1.0.4   gridGraphics_0.5-1  rlang_1.0.6         grid_4.1.2          rstudioapi_0.14    
[57] labeling_0.4.2      rmarkdown_2.18      gtable_0.3.1        DBI_1.1.3           R6_2.5.1            lubridate_1.9.0     knitr_1.41         
[64] fastmap_1.1.0       utf8_1.2.2          ragg_1.2.4          ape_5.6-2           stringi_1.7.8       parallel_4.1.2      Rcpp_1.0.9         
[71] vctrs_0.5.1         dbplyr_2.2.1        tidyselect_1.2.0    xfun_0.35  

This is the list of sMMO and pMMO biomarkers to screen against for the GtoTree workflow. This can then be brought into R to filter the summary table for hits using logical arguments where >=1 hit is required for core genes in the sMMO or pMMO pathway to flag genomes of interest. Use this list downstream to convert the headers for the summary of Pfam hits.tsv output by GtoTree.

**Note:** Could convert this to presence/absence table for all entries that have >1 or could the gene counts for each one.

# 1. Build the unrooted rooted tree for genomes from the Nevskiaceae family
1. Activate GtoTree and create a dedicated directory to house all the files you'll need for the tree. Copy MAGs of interest from your data to the directory.
```{bash}
source activate gtotree
mkdir ~/Nevskiaceae_phylogeny
cd ~/Nevskiaceae_phylogeny

mkdir GToTree_analyses
cd ~/Nevskiaceae_phylogeny/GToTree_analyses

mkdir Unrooted_tree

cp ~/directory/with/MAGs/MAG.fna ~/Nevskiaceae_phylogeny/GToTree_analyses/Unrooted_tree

cd ~/Nevskiaceae_phylogeny/GToTree_analyses/Unrooted_tree
ls *fna > Nevskiaceae_fasta_files.txt
```

2. Retrieve all the GTDB accession numbers for this family.
```{bash}
cd ~/Nevskiaceae_phylogeny/GToTree_analyses/Unrooted_tree
gtt-get-accessions-from-GTDB -t Nevskiaceae --GTDB-representatives-only
```
29 representative input genomes going in.

3. Create the list of Pfam entries for the sMMO and pMMO pathways + methanethiol oxidase to append to the tree-building step.
*Make sure to double check your entries at this stage*

*sMMO pathway (in approximate order of operon)*
**mmoX** PF02332
**mmoY** PF02332
**mmoB** PF02406
**mmoZ** PF02964
**mmoC** PF00970; PF00111; PF00175 (3 hits for subunits required for subcomponent C)
**mmoG** PF00118 (chaperon protein, not core to function)

*pMMO pathway (in order of operon)*
**pmoC** PF04896 
**pmoA** PF14100 (putative) or PF02461 (AmoA, more likely to occur)
**pmoB** PF04744 

*methanethiol oxidase (originally noticed in Mycobacteriaceae annotation)*
SELENBP1 PF05694
```{bash}
echo "PF02332" > Nevskiaceae_pfam_targets.txt #mmoX/mmoY same Pfam entry
echo "PF02406" >> Nevskiaceae_pfam_targets.txt #mmoB
echo "PF02964" >> Nevskiaceae_pfam_targets.txt #mmoZ
echo "PF00970" >> Nevskiaceae_pfam_targets.txt #mmoC 1/3
echo "PF00111" >> Nevskiaceae_pfam_targets.txt #mmoC 2/3
echo "PF00175" >> Nevskiaceae_pfam_targets.txt #mmoC 3/3
echo "PF00118" >> Nevskiaceae_pfam_targets.txt #mmoG
echo "PF04896" >> Nevskiaceae_pfam_targets.txt #pmoC/amoC
echo "PF14100" >> Nevskiaceae_pfam_targets.txt #pmoA putative
echo "PF02461" >> Nevskiaceae_pfam_targets.txt #pmoA/amoA
echo "PF04744" >> Nevskiaceae_pfam_targets.txt #pmoB/amoB
echo "PF05694" >> Nevskiaceae_pfam_targets.txt #SELENBP1 for methanethiol oxidase
cat Nevskiaceae_pfam_targets.txt #check the list
```

4. Run the GToTree script on the input genomes.
```{bash}
cd ~/Nevskiaceae_phylogeny/GToTree_analyses/Unrooted_tree

ls *fna > Nevskiaceae_fasta_files.txt

tmux new -s GtoTree_Nevskiaceae
tmux ls
tmux a -t GtoTree_Nevskiaceae

GToTree -f Nevskiaceae_fasta_files.txt\
        -a GTDB-Nevskiaceae-family-GTDB-rep-accs.txt \
        -p Nevskiaceae_pfam_targets.txt \
        -L Species \
        -H Bacteria -D -j 4 \
        -o Nevskiaceae_unrooted_tree
        
#Kill session if need be
tmux kill-session -t GtoTree_Nevskiaceae      
```

**Note:** From this point, you can export the tree files and metadata containing the Pfam hits that will be appended to the tree in R. 

# 2. Edit the tree metadata in R and visualize the tree
1. Bring the tab-delimited file containing Pfam hits for the different methane oxidizing proteins into R and edit the dataframe to create four categories. These categories are defined as sMMO (only hits for sMMO genes), pMMO (only hits for pMMO genes), sMMO_without_gamma (all hits for sMMO genes but missing the gamma hydrolase gene mmoZ), and sMMO_pMMO (genomes have hits for essential genes in both pathways).
```{r}
library(tidyverse)
library(glue)
Nevskiaceae_methane_Pfam_hits<- read.delim("Additional_Pfam_hit_counts.tsv", sep = "\t", header = TRUE) %>% 
                                select(-(X)) %>% #remove dummy column
                                rename(mmoX_mmoY = PF02332.20) %>% #rename columns to gene names
                                rename(mmoB = PF02406.19) %>%
                                rename(mmoZ = PF02964.18) %>%
                                rename(mmoC_1 = PF00970.26) %>%
                                rename(mmoC_2 = PF00111.29) %>%
                                rename(mmoC_3 = PF00175.23) %>%
                                rename(mmoG = PF00118.26) %>%
                                rename(pmoC_amoC = PF04896.14) %>%
                                rename(pmoA_putative = PF14100.8) %>%
                                rename(pmoA_amoA = PF02461.18) %>%
                                rename(pmoB_amoB = PF04744.14) %>%
                                rename(methanethiol_oxidase = PF05694.13)

Nevskiaceae_MMO_pathways_present<- Nevskiaceae_methane_Pfam_hits %>%
                                   mutate(Pathway_present = 
                                          case_when(
                                          mmoX_mmoY >=1 &
                                          mmoB >=1 &
                                          mmoZ >=1 &
                                          mmoC_1 >=1 &
                                          mmoC_2 >=1 &
                                          mmoC_3 >=1 & 
                                          (pmoC_amoC == 0 |
                                          pmoA_amoA == 0 |
                                          pmoB_amoB == 0) ~ "sMMO", #has all core genes for mmo and none for pmo
                                          
                                          mmoX_mmoY >=1 &
                                          mmoB >=1 &
                                          mmoZ == 0 & #code this to capture some without Gamma
                                          mmoC_1 >=1 &
                                          mmoC_2 >=1 &
                                          mmoC_3 >=1 & 
                                          (pmoC_amoC == 0 |
                                          pmoA_amoA == 0 |
                                          pmoB_amoB == 0) ~ "sMMO_without_gamma", #has most core genes for mmo, lacks mmoZ, 
                                          
                                          pmoC_amoC >=1 &
                                          pmoA_amoA >=1 & #include both versions here
                                          pmoB_amoB >=1 &
                                          (mmoX_mmoY == 0 | #define stringency for pMMO around absence of any part of A subcomponent
                                          #C subcomponent has multiple copies encoding it and seems to be a broad enzyme group with other fxn
                                          mmoB == 0 |
                                          mmoZ == 0) ~ "pMMO",
                                          
                                          mmoX_mmoY >=1 &
                                          mmoB >=1 &
                                          mmoZ >=1 &
                                          mmoC_1 >=1 &
                                          mmoC_2 >=1 &
                                          mmoC_3 >=1 & 
                                          pmoC_amoC >=1 &
                                          pmoA_amoA >=1 & #include both versions here
                                          pmoB_amoB >=1 ~ "sMMO_pMMO")) #%>%

#Filter out your MAG of interest, which will need to be handled separately, here we use MAG STE_114 to support the example      
Nevskiaceae_MMO_pathways_present_mydata<- Nevskiaceae_MMO_pathways_present %>% filter(assembly_id == "STE_114")

Nevskiaceae_metadata<-read.delim("GTDB-Nevskiaceae-family-GTDB-rep-metadata.tsv", sep = "\t", header = TRUE)

#left-join the data sets to get species names involved
Nevskiaceae_MMO_pathways_present <- Nevskiaceae_MMO_pathways_present %>%
                                         mutate(acss = str_replace(string = downloaded_accession,
                                                pattern = "GCA_",
                                                replacement = ""))

Nevskiaceae_metadata<- Nevskiaceae_metadata %>% 
                            mutate(acss = str_replace(string = accession,
                                                pattern = "GB_GCA_",
                                                replacement = "")) %>%
                            mutate(acss = str_replace(string = acss,
                                                pattern = "RS_GCF_",
                                                replacement = ""))


Nevskiaceae_MMO_pathways_present_taxonomy<- Nevskiaceae_metadata %>%
                                            select(acss, accession, genus, species) %>% #keep accessions so you can make this compatible with ggtree
                                            left_join(Nevskiaceae_MMO_pathways_present) %>%
                                            bind_rows(Nevskiaceae_MMO_pathways_present_mydata)


Nevskiaceae_MMO_pathways_present_taxonomy$acss[30] = "STE_114"
Nevskiaceae_MMO_pathways_present_taxonomy$accession[30] = "STE_114"
Nevskiaceae_MMO_pathways_present_taxonomy$species[30] = "STE_114"
Nevskiaceae_MMO_pathways_present_taxonomy$genus[30] = "STE_114"

Nevskiaceae_MMO_pathways_present_taxonomy<- Nevskiaceae_MMO_pathways_present_taxonomy %>%
                                            mutate(label = str_replace(string = accession,
                                                pattern = "GB_",
                                                replacement = "")) %>%
                                                mutate(label = str_replace(string = label,
                                                pattern = "RS_",
                                                replacement = "")) %>%
                                                mutate(species = str_replace(string = species, #add the underscore again to make compatible with tree
                                                pattern = " ",
                                                replacement = "_")) %>%
                                                mutate(label = as.factor(label)) %>%
                                                mutate(label = glue("{label}_{species}")) %>%
                                                mutate(label = str_replace(string = label,
                                                pattern = "_NA",
                                                replacement = "")) %>%
                                                relocate(label)

Nevskiaceae_MMO_pathways_present_taxonomy$label[30] = "STE_114"
```

2. Append the methane cycling metadata to the tree using colour coding.
```{r}
library(ggtree)
library(treeio)
library(tidyverse)
library(ggnewscale)
library(aplot)
library(tidytree)

tree <- read.tree("Nevskiaceae_unrooted_tree.tre")
circ <- ggtree(tree, layout = "circular", size=2, branch.length = "none", aes(x,y)) %<+%  
        Nevskiaceae_MMO_pathways_present_taxonomy +
        geom_tippoint(aes(color = Pathway_present), size = 10, show.legend = TRUE) +
        geom_tiplab(aes(label = str_replace(string =circ$data$species, pattern = "[_]", replacement = " "),
                    color = Pathway_present),
                    align = TRUE,
                    show.legend = FALSE,
                    size = 12) +
        scale_size(guide = "none") +
        scale_color_manual(name = "Pathway present",
                          values = c(
                          "pMMO" = "#CCCC00", 
                          "sMMO_pMMO" = "#00CC66", 
                          "sMMO_without_gamma" = "#333333",
                          "sMMO" = "#3399FF"), 
        labels = c("pMMO", "sMMO and pMMO", "sMMO, no gamma hydrolase", "sMMO"), #had to reorder because no hits for sMMO category in Nevskiaceae family
        na.translate = FALSE) +
        theme(legend.title = element_text(size = 32),
              legend.text = element_text(size = 30),
              legend.position = "bottom") +
        guides(color = guide_legend(override.aes = list(size=20))) +
        xlim_tree(20)
circ
ggsave("FigS6_Nevskiaceae_unrooted_phylogenetic_tree_MMO_pathway_presence.pdf", height = 25, width = 40, device = "pdf", limitsize = FALSE)
```
**Note:** This figure was subsequently manually edited in graphic vector software to add the legend item denoting sMMO and pMMO to align with Fig S6. In previous versions, we were able to manually add the sMMO category as a blue circle despite no genomes having a hit for that category but this seems to be deprecated since updating the packages and R studio software employed. I suspect this is also because of a ggtree update.